internal/controllers/machine/machine_controller_noderef_test.go

```diff
@@ -253,3 +257,79 @@ func TestGetManagedLabels(t *testing.T) {
 	got := getManagedLabels(allLabels)
 	g.Expect(got).To(BeEquivalentTo(managedLabels))
 }
+
+func TestReconcileNodeTaints(t *testing.T) {
+	tests := []struct {
+		name      string
+		node      *corev1.Node
+		wantError bool
+	}{
+		{
+			name: "if the node has the uninitialized taint it should be dropped from the node",
+			node: &corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-node-1",
+				},
+				Spec: corev1.NodeSpec{
+					Taints: []corev1.Taint{
+						clusterv1.NodeUninitializedTaint,
+					},
+				},
+			},
+			wantError: false,
+		},
+		{
+			name: "if the node does not have the uninitialized taint it should remain absent from the node",
+			node: &corev1.Node{
+				ObjectMeta: metav1.ObjectMeta{
+					Name: "test-node-1",
+				},
+				Spec: corev1.NodeSpec{
+					Taints: []corev1.Taint{},
+				},
+			},
+			wantError: false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			g := NewWithT(t)
+			fakeClient := fake.NewClientBuilder().WithObjects(tt.node).WithScheme(fakeScheme).Build()
+			r := &Reconciler{}
+			err := r.reconcileNodeTaints(ctx, fakeClient, tt.node)
+			if tt.wantError {
+				g.Expect(err).ToNot(BeNil())
+			} else {
+				g.Expect(err).To(BeNil())
+				g.Expect(tt.node.Spec.Taints).ShouldNot(ContainElement(MatchTaint(clusterv1.NodeUninitializedTaint)))
```

Comment:
```suggestion
				g.Expect(tt.node.Spec.Taints).ShouldNot(ContainElement(clusterv1.NodeUninitializedTaint))
```
This seems to work as well. WDYT?

Reply:
Ah you are right! Since we are not setting `Value` or `TimeAdded` on taints in the test cases `MatchTaint` (only compares `Key` and `Effect`) and `DeepEqual` become effectively equal.